Source/WebCore/platform/graphics/cairo/FontCairo.cpp

```diff
@@ -126,6 +127,30 @@ float Font::platformWidthForGlyph(Glyph glyph) const
     return width ? width : m_spaceWidth;
 }
 
+ResolvedEmojiPolicy FontCascade::resolveEmojiPolicy(FontVariantEmoji fontVariantEmoji, UChar32 character)
+{
+    switch (fontVariantEmoji) {
+    case FontVariantEmoji::Normal:
+    case FontVariantEmoji::Unicode:
+        // https://www.unicode.org/reports/tr51/#Presentation_Style
+        // There had been no clear line for implementers between three categories of Unicode characters:
+        // 1. emoji-default: those expected to have an emoji presentation by default, but can also have a text presentation
+        // 2. text-default: those expected to have a text presentation by default, but could also have an emoji presentation
+        // 3. text-only: those that should only have a text presentation
+        // These categories can be distinguished using properties listed in Annex A: Emoji Properties and Data Files.
+        // The first category are characters with Emoji=Yes and Emoji_Presentation=Yes.
+        // The second category are characters with Emoji=Yes and Emoji_Presentation=No.
+        // The third category are characters with Emoji=No.
+        if (u_hasBinaryProperty(character, UCHAR_EMOJI_PRESENTATION))
+            return ResolvedEmojiPolicy::RequireEmoji;
+        return ResolvedEmojiPolicy::NoPreference;
+    case FontVariantEmoji::Text:
+        return ResolvedEmojiPolicy::RequireText;
+    case FontVariantEmoji::Emoji:
+        return ResolvedEmojiPolicy::RequireEmoji;
+    }
+}
```

Comment:
Asking for help with this at https://webkit.slack.com/archives/CU65R5WQY/p1689189611828469